pss/outbox/outbox.go

```diff
@@ -0,0 +1,115 @@
+package outbox
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethersphere/swarm/log"
+	"github.com/ethersphere/swarm/pss/message"
+)
+
+type Config struct {
+	NumberSlots int
+	Forward     forwardFunction
+}
+
+type Outbox struct {
+	forwardFunc forwardFunction
+	queue       []*outboxMsg
+	slots       chan int
+	process     chan int
+	stopC       chan struct{}
+}
+
+type forwardFunction func(msg *message.Message) error
+
+var ErrOutboxFull = errors.New("outbox full")
```

Comment:
One open question to me is whether this is a good API for PSS - what is the advantage of returning an `outbox full` error to clients, versus for example blocking the clients? To me it seems like this makes the API harder to use for clients at no apparent gains.